src/cpp/rtps/xmlparser/XMLEndpointParser.cpp

```diff
@@ -249,6 +249,38 @@ XMLP_ret XMLEndpointParser::loadXMLReaderEndpoint(tinyxml2::XMLElement* xml_endp
                 return XMLP_ret::XML_ERROR;
             }
         }
+        else if (key == TOPIC) // deprecated api
+        {
+            const char *topicName = element->Attribute(NAME);
+            const char *typeName = element->Attribute(DATA_TYPE);
+            const char *kind = element->Attribute(KIND);
+
+            rdata->topicName() = topicName ? std::string(topicName) : std::string("");
+            rdata->typeName() = typeName ? std::string(typeName) : std::string("");
+            std::string auxString(kind ? kind : "");
+            if (auxString == _NO_KEY)
+            {
+                rdata->topicKind() = NO_KEY;
```

Comment:
```suggestion
                rdata->topicKind(NO_KEY);
```
As before, use setter and not getter method